### PR TITLE
Add initial support for setuptools in generated clients

### PR DESCRIPTION
**Pull request by coderfromhere**
Hi! I've implemented #3 from my work account.

I've used Nix to setup the dev environment, as I'm using multiple machines and it allows me to have a one-liner command to setup everything on a fresh hardware. I hope you don't mind it. It doesn't interfere with your `make bootstrap`.

This renderer API reminded me of [Cookiecutter](https://cookiecutter.readthedocs.io/en/1.7.2/). I have an [example template](https://github.com/avanov/python_project_template) that uses it for a generic python project , and I wonder if the same approach could be used here to offload project structure rendering to the third-party component.

### CODE REVIEW

**Comment by asyncee** [comment]
Everything looks good but Nix configuration. I think Nix is great piece of softwar, sadly i could not accept it because I have no experience with it and can not provide support for it to library users in the future, that is the main reason; also, i think there should be a single way to setup development environment so new users can easily dive in using well known way — currently devs are supposed to use `pyenv` and `poetry` tools which are well documented and widely used by python community.

Regarding cookiecutter — i have used it a lot and still using it in my projects. Currently i do not see the way to use it as a templating engine because of complex logic that is hidden inside library. I agree that top-level view on jinja2 and cookiecutter shows that this tools work in the same ways — both take context, template and render it.

**Comment by asyncee** [comment]
I'll try to test rest of the changes as soon as possible, hopefully tomorrow.

**Comment by coderfromhere** [comment]
ok, I removed those parts from the MR.

> also, i think there should be a single way to setup development environment so new users can easily dive in using well known way — currently devs are supposed to use pyenv and poetry tools which are well documented and widely used by python community.

Nix configuration was using Poetry under the hood. The difference was in the way it makes sure that Python and Poetry are available on the host machine. Currently there's no single way to bring Python and Poetry to the project dev env, users are supposed to take care of that themselves, and their experience may vary.

**Comment by asyncee** [comment]
Thanks! I will release new version soon.

**Comment by coderfromhere** [comment]
thank you!

**Comment by asyncee** [comment]
I released version 0.1.15. Thank you for your contribution!

Please, check readme to check latest changes (section about renderers).

tl;dr: i had to introduce separate renderer called `InstallablePackageRenderer` to preserve backward compatibility. You can specify new renderer in `.swagger-codegen.toml` like this:

```
renderer = "swagger_codegen.render.renderers.installable_package.InstallablePackageRenderer"
```

**Comment by coderfromhere** [comment]
What's the use case for a non-installable client?

**Comment by asyncee** [comment]
We have quite a few services where generated clients are part of a codebase, so i have to keep backwards compatibility. Anyway I think that installable clients may become a preferred approach.
